transport/rif_comms/node.py

```diff
@@ -1,135 +1,228 @@
-        print("RifCommsNode init on grpc endpoint: {}".format(self._config["grpc_endpoint"]))
-
-        self._greenlets: List[Greenlet] = list()
+        # initialize message queues
         self._address_to_message_queue: Dict[Address, MessageQueue] = dict()
 
-        self._stop_event = Event()
-        self._stop_event.set()
-
         self._log = log.bind(node_address=pex(self.address))
 
-    @property
-    def _queueids_to_queues(self) -> QueueIdsToQueues:
-        chain_state = views.state_from_raiden(self._raiden_service)
-        return views.get_all_messagequeues(chain_state)
-
     def start(self, raiden_service: RaidenService, message_handler: MessageHandler, prev_auth_data: str):
-        if not self._stop_event.ready():
+        """
+        Initialize transport fields, subscribe to own topic in order to listen for messages, and start as Runnable.
+        """
+        self._raiden_service = raiden_service  # TODO: this should be set in __init__
+
+        # check if node is already running
+        if not self.stop_event.ready():
             raise RuntimeError(f"{self!r} already started")
-        self._stop_event.clear()
-        self._raiden_service = raiden_service
+        self.stop_event.clear()
+
+        # connect to rif comms node
+        # TODO: this shouldn't need to be assigned, it is only done because otherwise the code hangs
+        self._rif_comms_connect_stream = self._comms_client.connect()
 
-        self._rif_comms_connect_stream = self._client.connect()
-        self._our_topic = self._client.subscribe(to_checksum_address(raiden_service.address))
-        self._client.get_peer_id(
-            to_checksum_address(raiden_service.address))  # TODO remove this after grpc api request blocking is fixed
+        # subscribe to our own topic to receive messages
+        self._start_message_listener()
+
+        # start pre-loaded message queues
         for message_queue in self._address_to_message_queue.values():
-            if not message_queue:
-                self.log.debug("Starting message_queue", message_queue=message_queue)
+            if not message_queue.greenlet:
+                self.log.debug("starting message_queue", message_queue=message_queue)
                 message_queue.start()
 
-        self.start_listener_thread()
-        self._greenlets = [self._our_topic_thread]
-        self.log.debug("RIF Comms Node started", config=self._config)
+        self.log.debug("RIF Comms Node start", config=self._config)
+
+        # start greenlet through the Runnable class; this will eventually call _run
         Runnable.start(self)
 
-    def __repr__(self):
-        node = f" node:{pex(self._raiden_service.address)}" if self._raiden_service else ""
+    def _start_message_listener(self):
+        """
+        Start a listener greenlet to listen for received messages in the background.
+        """
+        our_address = to_checksum_address(self.raiden_service.address)
```

Comment:
@marcosmartinez7 same here about the `to_checksum_address` call

Reply:
same, i dont think so, i think that addresses are already checksumed from when you select one on the cli

Reply:
removed